src/Coduo/PHPHumanizer/DateTime/PreciseFormatter.php

```diff
@@ -38,8 +41,20 @@ public function formatDifference(PreciseDifference $difference, $locale = 'en')
                 $locale
             );
         }
-        $suffix = $difference->isPast() ? 'compound.ago' : 'compound.from_now';
+        $suffix = $this->calculateCompoud($difference);
+        $suffixString = empty($suffix) ? '':' '.$suffix;
+
+        $prefix = $this->calculateCompoud($difference, '_prefix');
+        $prefixString = empty($prefix) ? '':$prefix.' ';
 
-        return implode(', ', $diff).' '.$this->translator->trans($suffix, array(), 'difference', $locale);
+        return $prefixString . implode(', ', $diff) . $suffixString;
```

Comment:
Any chance you could write some a tests for ``PreciseFormatter``? Now when we have also prefixes and suffixes it might be hard to understand this code.